### ISSUE TITLE
NuvolaAppSpotify: "Enable player in your browser" on Debian Jessie

### ISSUE THREAD

**Issue by gilcio** [tiliado/nuvolaruntime#462]
* Debian GNU/Linux 8.10 (jessie) 
* Desktop Environment: XFCE 
* Spotify script  Version: 3.0.2  Revision: 2-gbdb2b60 
* Nuvola Apps - Genuine flatpak build  Version 4.12.0  Revision 0-g705ce93 
* Diorite 4.12.0 WebKitGTK 2.20.3 Chromium 67.0.3396.79  libsoup 2.62.2 

what steps you performed 
- Started Spotify app. 

what happened 

- Message "Enable player in your browser" 
![spotify_enable_player_error](https://user-images.githubusercontent.com/41997134/43520486-6a68f54e-9569-11e8-93ce-2aa72c190c47.png)

what you expected 
- Spotify ready to use. 

P.S.: I can use other apps from Nuvola, like Deezer, Pandora, TuneIn, Youtube, without any problem. 

```
$ flatpak run eu.tiliado.NuvolaAppSpotify --debug 
Runner: [DEBUG Δ028002us Nuvola] AboutDialog.vala:54: Nuvola Apps 4.12.0 (0-g705ce93) 
Runner: [DEBUG Δ004000us Nuvola] AboutDialog.vala:56: Spotify script 3.0.2 (2-gbdb2b60) 
Runner: [DEBUG Δ000000us Nuvola] AboutDialog.vala:60: Diorite 4.12.0 
Runner: [DEBUG Δ000000us Nuvola] AboutDialog.vala:62: Chromium 67.0.3396.79 
Runner: [DEBUG Δ044003us DioriteGtk] DesktopShell.vala:251: Shell: xfwm4 (null), CSD 0, appmenu 0, menubar 0 
Runner: [DEBUG Δ024001us Nuvola] Connection.vala:115: Network Proxy: system settings 
Runner: [DEBUG Δ056004us GdkPixbuf] gdk_pixbuf_from_pixdata() called on: 
Runner: [DEBUG Δ000000us GdkPixbuf]         Encoding raw 
Runner: [DEBUG Δ000000us GdkPixbuf]         Dimensions: 14 x 14 
Runner: [DEBUG Δ000000us GdkPixbuf]         Rowstride: 56, Length: 808 
Runner: [DEBUG Δ000000us GdkPixbuf]         Copy pixels == false 
Runner: [DEBUG Δ056003us Nuvola] AppRunnerController.vala:116: Scale factor: 1 
Runner: [DEBUG Δ012001us Nuvola] FormatSupport.vala:86: Trying to play /app/share/nuvolaruntime/audio/audiotest.mp3. 
Runner: [DEBUG Δ072005us DioriteGtk] Application.vala:243: XFCE session appeared: org.xfce.SessionManager, :1.0 
Runner: [DEBUG Δ004000us Nuvola] FormatSupport.vala:86: Pipeline state changed from NULL to READY. 
Runner: [DEBUG Δ000000us Nuvola] FormatSupport.vala:86: Pipeline state changed from READY to PAUSED. 
Runner: [DEBUG Δ004000us Nuvola] FormatSupport.vala:86: Pipeline state changed from PAUSED to PLAYING. 
Runner: [DEBUG Δ000000us Nuvola] FormatSupport.vala:86: End of stream for file /app/share/nuvolaruntime/audio/audiotest.mp3. 
Runner: [DEBUG Δ008001us GdkPixbuf] gdk_pixbuf_from_pixdata() called on: 
Runner: [DEBUG Δ000000us GdkPixbuf]         Encoding raw 
Runner: [DEBUG Δ000000us GdkPixbuf]         Dimensions: 14 x 14 
Runner: [DEBUG Δ000000us GdkPixbuf]         Rowstride: 56, Length: 808 
Runner: [DEBUG Δ000000us GdkPixbuf]         Copy pixels == false 
Runner: [DEBUG Δ016001us GdkPixbuf] gdk_pixbuf_from_pixdata() called on: 
Runner: [DEBUG Δ000000us GdkPixbuf]         Encoding raw 
Runner: [DEBUG Δ000000us GdkPixbuf]         Dimensions: 14 x 14 
Runner: [DEBUG Δ000000us GdkPixbuf]         Rowstride: 56, Length: 808 
Runner: [DEBUG Δ000000us GdkPixbuf]         Copy pixels == false 
Runner: [DEBUG Δ004000us Nuvola] StartupCheck.vala:216: Checking requirements with Chromium 67.0.3396.79 
Runner: [DEBUG Δ000000us Nuvola] Traits.vala:62: Chromium[63] -> DRT_REQUIREMENT_STATE_SUPPORTED 
Runner: [DEBUG Δ000000us Nuvola] Traits.vala:62: Feature[widevine] -> DRT_REQUIREMENT_STATE_UNKNOWN 
Runner: [DEBUG Δ000000us Nuvola] Traits.vala:62: Feature[MSE] -> DRT_REQUIREMENT_STATE_SUPPORTED 
Runner: [DEBUG Δ000000us CefGtk] WidevinePlugin.vala:15: Widevine path: /home/user/.var/app/eu.tiliado.NuvolaAppSpotify/data/nuvolaruntime/apps_data/spotify/widevine 
Your system is broken: dlsym doesn't work! This has been reported to be caused by Nvidia's libGL. You should expect time related functions to misbehave. http://code.google.com/p/chromium/issues/detail?id=16800 
[0731/161239.710863:ERROR:address_tracker_linux.cc(175)] Could not bind NETLINK socket: Address already in use (98) 
Runner: [DEBUG Δ084005us GdkPixbuf] gdk_pixbuf_from_pixdata() called on: 
Runner: [DEBUG Δ008001us GdkPixbuf]         Encoding raw 
Runner: [DEBUG Δ000000us GdkPixbuf]         Dimensions: 14 x 14 
Runner: [DEBUG Δ000000us GdkPixbuf]         Rowstride: 56, Length: 808 
Runner: [DEBUG Δ000000us GdkPixbuf]         Copy pixels == false 
Runner: [INFO Δ016001us CefGtk] WidevinePlugin.vala:27: WidevinePlugin is available: 0 (null) 
Runner: [DEBUG Δ000000us Nuvola] StartupCheck.vala:247: Checking requirements with Chromium 67.0.3396.79 
Runner: [DEBUG Δ000000us Nuvola] Traits.vala:62: Chromium[63] -> DRT_REQUIREMENT_STATE_SUPPORTED 
Runner: [DEBUG Δ000000us Nuvola] Traits.vala:62: Feature[widevine] -> DRT_REQUIREMENT_STATE_SUPPORTED 
Runner: [DEBUG Δ000000us Nuvola] Traits.vala:62: Feature[MSE] -> DRT_REQUIREMENT_STATE_SUPPORTED 
Runner: [WARNING Δ020001us Nuvola] StartupWindow.vala:205: tiliado-account-message: No Tiliado account. 
Runner: [DEBUG Δ016001us Nuvola] WebkitOptions.vala:51: MSE supported: no 
Runner: [INFO Δ000000us Nuvola] CefEngine.vala:298: Partially implemented: register_ipc_handlers() 
Runner: [INFO Δ004000us CefGtk] WebView.vala:74: CEF rendering mode: CEF_GTK_RENDERING_MODE_WINDOWED 
Runner: [INFO Δ072005us CefGtk] Client.vala:106: get_life_span_handler 
Runner: [DEBUG Δ032002us CefGtk] WebViewWindowed.vala:65: allocation 0,1+1100,508 child_allocation 0,0+1100,508 
Runner: [INFO Δ060003us CefGtk] WebView.vala:113: Renderer #1 created. 
Runner: [DEBUG Δ084006us Nuvola] WebAppWindow.vala:180: Sidebar position: 863 
[0731/161240.070886:ERROR:gpu_process_transport_factory.cc(1027)] Lost UI shared context. 
Runner: [DEBUG Δ056003us Nuvola] CefEngine.vala:374: Init finished 
Runner: [DEBUG Δ000000us Nuvola] WebWorker.vala:80: Cannot call ('Nuvola.actions.emit', <('ActionEnabledChanged', 'go-forward', false)>, false) 
Runner: [DEBUG Δ000000us Nuvola] WebWorker.vala:80: Cannot call ('Nuvola.actions.emit', <('ActionEnabledChanged', 'go-back', false)>, false) 
Runner: [DEBUG Δ008001us Nuvola] AppRunnerController.vala:697: Component tray_icon (Tray Icon) available=true, enabled=false 
Runner: [DEBUG Δ000000us Nuvola] AppRunnerController.vala:697: Component audio_tweaks (Audio Tweaks (beta)) available=true, enabled=false 
Runner: [INFO Δ000000us Nuvola] Component.vala:59: Load unity_launcher Dock Menu Items 
Runner: [INFO Δ000000us Nuvola] Component.vala:109: Activate: unity_launcher Dock Menu Items 
Runner: [DEBUG Δ004000us Nuvola] AppRunnerController.vala:697: Component unity_launcher (Dock Menu Items) available=true, enabled=true 
Runner: [DEBUG Δ000000us Nuvola] AppRunnerController.vala:697: Component notifications (Notifications) available=true, enabled=false 
Runner: [INFO Δ000000us Nuvola] Component.vala:59: Load mediakeys Media keys 
Runner: [DEBUG Δ000000us Nuvola] AppRunnerController.vala:697: Component mediakeys (Media keys) available=true, enabled=true 
Runner: [INFO Δ000000us Nuvola] Component.vala:59: Load scrobbler Audio Scrobbler Services 
Runner: [DEBUG Δ004000us Nuvola] AppRunnerController.vala:697: Component scrobbler (Audio Scrobbler Services) available=true, enabled=true 
Runner: [INFO Δ000000us Nuvola] Component.vala:59: Load mpris Media Player DBus Interface (MPRIS 2) 
Runner: [DEBUG Δ000000us Nuvola] AppRunnerController.vala:697: Component mpris (Media Player DBus Interface (MPRIS 2)) available=true, enabled=true 
Runner: [DEBUG Δ000000us Nuvola] AppRunnerController.vala:697: Component httpremotecontrol (Remote control over HTTP (experimental)) available=true, enabled=false 
Runner: [INFO Δ000000us Nuvola] Component.vala:59: Load lyrics Lyrics 
Runner: [DEBUG Δ000000us Nuvola] AppRunnerController.vala:697: Component lyrics (Lyrics) available=true, enabled=true 
Runner: [INFO Δ000000us Nuvola] Component.vala:59: Load developer Developer's tools 
Runner: [INFO Δ000000us Nuvola] Component.vala:109: Activate: developer Developer's tools 
[0731/161240.146891:ERROR:command_buffer_proxy_impl.cc(133)] ContextResult::kTransientFailure: Failed to send GpuChannelMsg_CreateCommandBuffer. 
Worker: [DEBUG Δ108007us Nuvola] CefRendererExtension.vala:137: Got JS context 
[0731/131240.166892:WARNING:ipc_message_attachment_set.cc(49)] MessageAttachmentSet destroyed with unconsumed attachments: 0/1 
Worker: [INFO Δ012001us CefGtk] RendererContext.vala:57: JS Context created 1 
Runner: [DEBUG Δ056004us Nuvola] WebWorker.vala:80: Cannot call ('Nuvola.actions.emit', <('ActionEnabledChanged', 'toggle-sidebar', true)>, false) 
Runner: [DEBUG Δ004000us Nuvola] WebWorker.vala:80: Cannot call ('Nuvola.actions.emit', <('ActionEnabledChanged', 'toggle-sidebar', true)>, false) 
Runner: [DEBUG Δ012001us Nuvola] AppRunnerController.vala:697: Component developer (Developer's tools) available=true, enabled=true 
Runner: [DEBUG Δ020001us Nuvola] CefEngine.vala:160: App Runner Initialized 
Runner: [DEBUG Δ000000us Nuvola] CefEngine.vala:164: App Runner Ready 
Runner: [DEBUG Δ008000us Nuvola] CefEngine.vala:377: Web Worker Ready 
Runner: [DEBUG Δ024002us Nuvola] CefEngine.vala:585: Navigation: new window: false → false, uri: 'https://open.spotify.com/browse/featured' → 'https://open.spotify.com/browse/featured', approved: true, frame: (null), type: TT_EXPLICIT, user gesture: true 
Runner: [DEBUG Δ008000us Nuvola] CefEngine.vala:484: Load started 

(eu.tiliado.NuvolaAppSpotify:3): Gtk-WARNING **: 13:12:40.274: Unable to parse accelerator '': ignored request to install 1 accelerators 

(eu.tiliado.NuvolaAppSpotify:3): Gtk-WARNING **: 13:12:40.274: Unable to parse accelerator '': ignored request to install 1 accelerators 
Runner: [DEBUG Δ024002us Nuvola] CefEngine.vala:484: Load started 
Runner: [INFO Δ000000us Nuvola] Component.vala:109: Activate: mpris Media Player DBus Interface (MPRIS 2) 
Runner: [INFO Δ000000us Nuvola] Component.vala:109: Activate: lyrics Lyrics 
Runner: [DEBUG Δ004000us Nuvola] WebWorker.vala:80: Cannot call ('Nuvola.actions.emit', <('ActionEnabledChanged', 'toggle-sidebar', true)>, false) 
Runner: [INFO Δ000000us Nuvola] Component.vala:109: Activate: mediakeys Media keys 
Runner: [DEBUG Δ000000us Nuvola] MPRISProvider.vala:67: Bus acquired: org.mpris.MediaPlayer2.NuvolaAppSpotify, registering objects 
Runner: [CRITICAL Δ000000us Nuvola] MPRISProvider.vala:85: Bus name lost: org.mpris.MediaPlayer2.NuvolaAppSpotify 
Runner: [CRITICAL Δ000000us Nuvola] Application will not function properly. 
Runner: [INFO Δ004000us Nuvola] Component.vala:109: Activate: scrobbler Audio Scrobbler Services 
Worker: [DEBUG Δ01s032064us Nuvola] CefRendererExtension.vala:145: Lost JS context 
Worker: [INFO Δ000000us CefGtk] RendererContext.vala:64: JS Context released: 1 
Worker: [DEBUG Δ016001us Nuvola] CefRendererExtension.vala:137: Got JS context 
Worker: [INFO Δ032002us CefGtk] RendererContext.vala:57: JS Context created 1 
Runner: [DEBUG Δ980062us Nuvola] CefEngine.vala:484: Load started 
Runner: [DEBUG Δ308019us Nuvola] DeveloperSidebar.vala:193: Media player notify: can-seek 
Runner: [DEBUG Δ000000us Nuvola] DeveloperSidebar.vala:193: Media player notify: can-change-volume 
Runner: [DEBUG Δ000000us Nuvola] DeveloperSidebar.vala:193: Media player notify: can-go-previous 
Runner: [DEBUG Δ004000us Nuvola] DeveloperSidebar.vala:193: Media player notify: can-go-next 
Runner: [DEBUG Δ000000us Nuvola] DeveloperSidebar.vala:193: Media player notify: can-play 
Runner: [DEBUG Δ004000us Nuvola] DeveloperSidebar.vala:193: Media player notify: can-pause 
Runner: [DEBUG Δ000000us Nuvola] DeveloperSidebar.vala:193: Media player notify: can-stop 
[0731/161241.638984:WARNING:url_request_job_manager.cc(90)] Failed to map: chrome-extension://invalid/ 
[0731/161241.646984:WARNING:url_request_job_manager.cc(90)] Failed to map: chrome-extension://invalid/ 
[0731/161242.275024:WARNING:value_store_frontend.cc(42)] Reading mhjfbmdgcfjbbpaeojofohoefgiehjai.alarms from  failed: IO error: .../LOCK: No further details. (ChromeMethodBFE: 15::LockFile::1) 
Runner: [INFO Δ808051us CefGtk] WebView.vala:105: Console: https://www.gstatic.com/cast/sdk/libs/sender/1.0/cast_framework.js:37: Uncaught TypeError: Cannot read property 'AutoJoinPolicy' of undefined 
[0731/161242.819058:WARNING:backend_impl.cc(1832)] Destroying invalid entry. 
[0731/161245.579230:WARNING:backend_impl.cc(1832)] Destroying invalid entry. 
[0731/161246.527289:ERROR:ssl_client_socket_impl.cc(1026)] handshake failed; returned -1, SSL error code 1, net_error -101 
Runner: [INFO Δ04s148259us CefGtk] WebView.vala:105: Console: https://open.scdn.co/static/web-player.456e177d.js:1: WebSocket connection to 'wss://guc-dealer.spotify.com/?access_token=BQAtCxvVthGu8R0cV4TSMD7iEtwR8UhuhieAlGWHbMlS5M2D5Ug54NfWxbYkXlGLsmKlyojEIiOD8mfjEoGxHjpEfIVPw0saRZ0U_GERW-CxHqN_UBHQ74bI5HMO2CAbOfZ5k-DJM4YUR6aLXPu420DIH9LilfW1jp0fxBhjrrLDqYNBKkXSKg-jqStco2eyB76rnlg0z9LqfghdnXzuc_-FMapHlulBrtdp3fPU7pAW7RrgpScyZg485R3MPboHfx6iU1tXKFSMsh0' failed: Error in connection establishment: net::ERR_CONNECTION_RESET 
[0731/161247.447347:WARNING:backend_impl.cc(1832)] Destroying invalid entry. 
Runner: [INFO Δ01m03s283955us CefGtk] Client.vala:106: get_life_span_handler 
```

**Comment by fenryxo** [comment]
Hello @gilcio. It might be an issue with Widevine plugin. What does the following command show?

```
cat ~/.var/app/eu.tiliado.NuvolaAppSpotify/data/nuvolaruntime/apps_data/spotify/widevine/version.txt
```

Does it help if you delete the plugin and launch Spotify to install it again (please collect and attach the debug log of this as a file - there is a link "Attach files by dragging & dropping or selecting them" under the comment text box)?

```
rm -r ~/.var/app/eu.tiliado.NuvolaAppSpotify/data/nuvolaruntime/apps_data/spotify/widevine
flatpak run eu.tiliado.NuvolaAppSpotify -D
```

**Comment by fenryxo** [comment]
I get `Widevine CDM registration failed; Manifest has no supported x-cdm-interface-versions in '8'` with `Chromium 68.0.3440.75`.

This manifest seems to work:

```
{
  "os": "linux",
  "arch": "x64",
  "x-cdm-module-versions": "4",
  "x-cdm-interface-versions": "8,9",
  "x-cdm-host-versions": "8,9",
  "x-cdm-codecs": "vp8,vp9.0,avc1",
  "version":  "1.4.8.903"
}
```

**Comment by gilcio** [comment]
Here's the debug log file:
[output.txt](https://github.com/tiliado/nuvolaruntime/files/2262387/output.txt)

Also, I tried deleting the broken link ~/.var/app/eu.tiliado.NuvolaAppSpotify/data/nuvolaruntime/apps_data/spotify/widevine/manifest.json and creating a regular file with the content from the quoted text in your last post, but no luck.

**Comment by fenryxo** [comment]
> Also, I tried deleting the broken link ~/.var/app/eu.tiliado.NuvolaAppSpotify/data/nuvolaruntime/apps_data/spotify/widevine/manifest.json

The link is not broken but it points to a file that is available under that location only in the sandbox (the location outside the snadbox is /var/lib/flatpak/app/eu.tiliado.NuvolaAppSpotify/x86_64/stable/active/files/lib/cef/manifest.json).

> and creating a regular file with the content from the quoted text in your last post, but no luck.

Could you try again with the development version (4.12.11 from the master branch) I've been testing?

```
$ flatpak run eu.tiliado.NuvolaAppSpotify//stable --version
[...]
Version 4.12.0
Revision 0-g705ce93
[...]
Chromium 67.0.3396.79
$ flatpak install nuvola eu.tiliado.NuvolaAppSpotify//master
$ flatpak run eu.tiliado.NuvolaAppSpotify//master --version
[...]
Version 4.12.11
Revision 11-ge122dd8
[...]
Chromium 68.0.3440.75
$ flatpak run eu.tiliado.NuvolaAppSpotify//master -D
```

**Comment by gilcio** [comment]
Now it stops on the "Start-up Check for Spotify" screen displaying error in "Web App Requirements".
[output.txt](https://github.com/tiliado/nuvolaruntime/files/2262626/output.txt)

**Comment by fenryxo** [comment]
> Widevine CDM registration failed; Manifest has no supported x-cdm-interface-versions in '8'

It reads Nuvola's old `manifest.json` (the flatpaks don't contain the new one yet). Can you verify that `~/.var/app/eu.tiliado.NuvolaAppSpotify/data/nuvolaruntime/apps_data/spotify/widevine/manifest.json` is still your regular file with the new content?

**Comment by gilcio** [comment]
Sorry, I forgot to put the manifest file again. 
Now it's displaying a warning about Nuvola Service, but after clicking "Continue" the app opens and works properly!
Is there a way to skip this step?
[output.txt](https://github.com/tiliado/nuvolaruntime/files/2262860/output.txt)
![warning](https://user-images.githubusercontent.com/41997134/43721623-ad77233e-9969-11e8-99a1-0998117aac8f.png)

**Comment by fenryxo** [comment]
This warning will disappear when the fix propagates to the stable builds and you switch back to the stable flatpak of Spotify. As a workaround, you can install the development version of Nuvola Service and start it before launching Spotify.

```
flatpak install nuvola eu.tiliado.Nuvola//master
flatpak run --command=nuvola eu.tiliado.Nuvola//master
```

Note that this is a temporary workaround until the fix gets to the stable channel (possibly tomorrow). I highly discourage you from running development builds as a default because they may sometimes not work correctly or at all.

**Comment by fenryxo** [comment]
Or you can uninstall Nuvola service entirely (but some features depend on it).

**Comment by gilcio** [comment]
That's right fenryxo! Thank you so much for your support!

**Comment by fenryxo** [comment]
Please keep this issue open until the fix is released.

**Comment by fenryxo** [comment]
Released in Nuvola 4.12.20: [Nuvola Apps Runtime Fixes Issue with Spotify & Widevine Plugin, Tweaks Dark Theme Option and Scrollbars](https://medium.com/nuvola-news/nuvola-apps-runtime-fixes-issue-with-spotify-widevine-plugin-tweaks-dark-theme-option-and-dbd22ebdce04)
